modules/xmpp/JingleSessionPC.js

```diff
@@ -242,48 +243,54 @@ export default class JingleSessionPC extends JingleSession {
     }
 
     /**
-     *
+     * @inheritDoc
+     * @param {JingleSessionPCOptions} options  - a set of config options.
      */
-    doInitialize() {
+    doInitialize(options) {
+        this.failICE = Boolean(options.failICE);
         this.lasticecandidate = false;
+        this.options = options;
 
-        // True if reconnect is in progress
+        /**
+         * {@code true} if reconnect is in progress.
+         * @type {boolean}
+         */
         this.isReconnect = false;
 
-        // Set to true if the connection was ever stable
+        /**
+         * Set to {@code true} if the connection was ever stable
+         * @type {boolean}
+         */
         this.wasstable = false;
+        this.webrtcIceUdpDisable = Boolean(options.webrtcIceUdpDisable);
+        this.webrtcIceTcpDisable = Boolean(options.webrtcIceTcpDisable);
 
-        const pcOptions = { disableRtx: this.room.options.disableRtx };
+        const pcOptions = { disableRtx: options.disableRtx };
 
-        if (this.room.options.gatherStats) {
+        if (options.gatherStats) {
             pcOptions.maxstats = DEFAULT_MAX_STATS;
         }
 
         if (this.isP2P) {
             // simulcast needs to be disabled for P2P (121) calls
             pcOptions.disableSimulcast = true;
-            pcOptions.disableH264
-                = this.room.options.p2p && this.room.options.p2p.disableH264;
-            pcOptions.preferH264
-                = this.room.options.p2p && this.room.options.p2p.preferH264;
+            pcOptions.disableH264 = options.p2p && options.p2p.disableH264;
+            pcOptions.preferH264 = options.p2p && options.p2p.preferH264;
 
-            const abtestSuspendVideo = this._abtestSuspendVideoEnabled();
+            const abtestSuspendVideo = this._abtestSuspendVideoEnabled(options);
 
             if (typeof abtestSuspendVideo !== 'undefined') {
                 pcOptions.abtestSuspendVideo = abtestSuspendVideo;
             }
         } else {
             // H264 does not support simulcast, so it needs to be disabled.
             pcOptions.disableSimulcast
-                = this.room.options.disableSimulcast
-                    || (this.room.options.preferH264
-                            && !this.room.options.disableH264);
-            pcOptions.preferH264 = this.room.options.preferH264;
+                = options.disableSimulcast
+                    || (options.preferH264 && !options.disableH264);
+            pcOptions.preferH264 = options.preferH264;
             pcOptions.enableFirefoxSimulcast
-                = this.room.options.testing
-                    && this.room.options.testing.enableFirefoxSimulcast;
-            pcOptions.enableLayerSuspension
-                = this.room.options.enableLayerSuspension;
+                = options.testing && options.testing.enableFirefoxSimulcast;
+            pcOptions.enableLayerSuspension = options.enableLayerSuspension;
```

Comment:
Not sure if it has any impact, but this is not going to work if there is no simulcast, is it? Maybe check for simulcast too then?

Reply:
This commit refactors the options, but I don't want to do any changes to the logic behind it.